packages/@aws-cdk/aws-codepipeline/test/notifications.test.ts

```diff
@@ -0,0 +1,206 @@
+            {
+              Ref: 'PipelineC660917D',
+            },
+          ],
+        ],
+      },
+      Targets: [
+        {
+          TargetAddress: 'SlackID',
+          TargetType: 'AWSChatbotSlack',
+        },
+      ],
+    });
+  });
+
+  test('On "Pipeline execution" events emitted notification rule', () => {
+    pipeline.notifyOnPipelineStateChange('NotifyOnPipelineStateChange', { target: { bind: () => ({ targetType: notifications.TargetType.AWS_CHATBOT_SLACK, targetAddress: 'SlackID' }) } });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'PipelineNotifyOnPipelineStateChange37743E22',
+      DetailType: 'FULL',
+      EventTypeIds: [
+        'codepipeline-pipeline-pipeline-execution-failed',
+        'codepipeline-pipeline-pipeline-execution-canceled',
+        'codepipeline-pipeline-pipeline-execution-started',
+        'codepipeline-pipeline-pipeline-execution-resumed',
+        'codepipeline-pipeline-pipeline-execution-succeeded',
+        'codepipeline-pipeline-pipeline-execution-superseded',
+      ],
+      Resource: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:',
+            {
+              Ref: 'AWS::Partition',
+            },
+            ':codepipeline:',
+            {
+              Ref: 'AWS::Region',
+            },
+            ':',
+            {
+              Ref: 'AWS::AccountId',
+            },
+            ':',
+            {
+              Ref: 'PipelineC660917D',
+            },
+          ],
+        ],
+      },
+      Targets: [
+        {
+          TargetAddress: 'SlackID',
+          TargetType: 'AWSChatbotSlack',
+        },
+      ],
+    });
+  });
+
+  test('On "Manual approval" events emitted notification rule', () => {
+    pipeline.notifyOnApprovalStateChange('NotifyOnApprovalStateChange', { target: { bind: () => ({ targetType: notifications.TargetType.AWS_CHATBOT_SLACK, targetAddress: 'SlackID' }) } });
+
+    expect(stack).toHaveResourceLike('AWS::CodeStarNotifications::NotificationRule', {
+      Name: 'PipelineNotifyOnApprovalStateChange919F4E17',
+      DetailType: 'FULL',
+      EventTypeIds: [
+        'codepipeline-pipeline-manual-approval-failed',
+        'codepipeline-pipeline-manual-approval-needed',
+        'codepipeline-pipeline-manual-approval-succeeded',
+      ],
+      Resource: {
+        'Fn::Join': [
+          '',
+          [
+            'arn:',
+            {
+              Ref: 'AWS::Partition',
+            },
+            ':codepipeline:',
+            {
+              Ref: 'AWS::Region',
+            },
+            ':',
+            {
+              Ref: 'AWS::AccountId',
+            },
+            ':',
+            {
+              Ref: 'PipelineC660917D',
+            },
+          ],
+        ],
+      },
+      Targets: [
+        {
+          TargetAddress: 'SlackID',
+          TargetType: 'AWSChatbotSlack',
+        },
+      ],
+    });
+  });
+});
```

Comment:
All of the comments I left on the CodeBuild module changes apply to the changes on the CodePipeline module as well.

Reply:
Done.